src/api/java/com/minecolonies/api/research/IGlobalResearch.java

```diff
@@ -123,12 +163,36 @@
      */
     void addChild(IGlobalResearch child);
 
+    /**
+     * Add a child to a research, without setting parentage.
+     * @param child the child to add
+     */
+    void addChild(final String child);
+
+    /**
+     * Add an individual cost.
+     * @param cost the individual itemstack to add to the cost list, as a string.
+     */
+    void addCost(final String cost);
```

Comment:
Costs are ItemStacks, including quantity (and, while not currently supported, down the road possibly NBT) information.  These ones I can pre-format into their ItemStacks if that's preferred, but it will make the serialization/deserialization much worse from a performance perspective.

Reply:
Then make it item stacks

Or something that generally represents itemstacks or other objects like an interface implemented on wrappers.

We are in an object oriented language with java. Make use of it and it's compiler to prevent runtime problems.